app/src/main/java/com/T05/krowdtrialz/ui/ExperimentDetailsNonOwnerActivity.java

```diff
@@ -49,58 +56,108 @@ protected void onCreate(Bundle savedInstanceState) {
         // TODO: Use this to get experiment object
         String experimentID = intent.getStringExtra(SubscribedFragment.EXTRA_EXPERIMENT_ID);
 
+        this.experiment = experiment;
```

Comment:
I think this line is just referencing the experiment object to itself. You might have to do this instead:
`this.experiment = Database.getInstance().getExperimentByID(experimentID);`

Reply:
Ah, this was meant to be removed with the testing code that just hard coded an experiment. I will change it now. Thanks